receiver/zipkinreceiver/trace_receiver.go

```diff
@@ -288,33 +294,61 @@ const (
 // unmarshals them and sends them along to the nextConsumer.
 func (zr *ZipkinReceiver) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// Trace this method
-	ctx, span := trace.StartSpan(context.Background(), "ZipkinReceiver.Export")
+	parentCtx := r.Context()
+	ctx, span := trace.StartSpan(parentCtx, "ZipkinReceiver.Export")
 	defer span.End()
 
 	// If the starting RPC has a parent span, then add it as a parent link.
-	// TODO: parentCtx should be direct parent for the span created here.
-	parentCtx := r.Context()
 	observability.SetParentLink(parentCtx, span)
 
+	// Now deserialize and process the spans.
+	asZipkinv1 := r.URL != nil && strings.Contains(r.URL.Path, "api/v1/spans")
+
+	var receiverTagValue string
+	if asZipkinv1 {
+		receiverTagValue = zipkinV1TagValue
+	} else {
+		receiverTagValue = zipkinV2TagValue
+	}
+
+	ctxWithReceiverName := observability.ContextWithReceiverName(ctx, receiverTagValue)
+
+	if !zr.host.OkToIngest() {
+		var responseStatusCode int
+		var zPageMessage string
+		if zr.backPressureState == configmodels.EnableBackPressure {
+			responseStatusCode = http.StatusServiceUnavailable
```

Comment:
In the future we can also try to be smart and calculate queue drain speed and the time it will be required for it to drain to certain reasonable level and report that in "Retry-After" header. Not for this PR, just something to consider for later.